code/game/machinery/kitchen/microwave.dm

```diff
@@ -416,40 +401,47 @@
 	return ffuu
 
 /obj/machinery/microwave/Topic(href, href_list)
+	SSvueui.check_uis_for_change(src)
 	if(..())
 		return
 
 	usr.set_machine(src)
+
 	if(src.operating)
-		src.updateUsrDialog()
+		if(href_list["abort"])
+			abort = TRUE
+		SSvueui.check_uis_for_change(src)
 		return
 
-	switch(href_list["action"])
-		if ("cook")
-			cook()
+	if(href_list["cook"])
+		cook()
+	else if(href_list["eject_all"])
+		eject()
+	else if(href_list["eject"])
+		for (var/datum/reagent/R in src.reagents.reagent_list)
+			if(R.name == href_list["eject"])
+				if(istype(usr.l_hand, /obj/item/weapon/reagent_containers) || istype(usr.r_hand, /obj/item/weapon/reagent_containers))
+					var/obj/item/weapon/reagent_containers/RC = usr.l_hand || usr.r_hand
+					var/vol_diff = R.volume - RC.reagents.get_free_space()
+					if(vol_diff > R.volume)
+						to_chat(usr, "<span class='notice'>You empty [R.volume] units of [R.name] into your [RC.name].</span>")
+						RC.reagents.add_reagent(R.id, R.volume)
+						src.reagents.remove_reagent(R.id, R.volume)
+					else
+						to_chat(usr, "<span class='notice'>You empty [vol_diff] units of [R.name] into your [RC.name].</span>")
+						RC.reagents.add_reagent(R.id, vol_diff)
+						src.reagents.remove_reagent(R.id, vol_diff)
+					SSvueui.check_uis_for_change(src)
+					break
+				else
+					to_chat(usr, "<span class='warning'>You need to be holding a valid container to empty [R.name]!</span>")

```

Comment:
I mean... how are they gonna empty something that isnt a container?

Reply:
It's an error message for trying to empty reagents out of the microwave without a container in their hand, not for trying to empty not-a-container into the microwave.